File: src/domain/stack-pack-mapping.ts
/**
 * Framework to knowledge pack mapping.
 *
 * Migrated from Python `domain/stack_pack_mapping.py`.
 */

/** Framework name to knowledge pack directory name (11 entries). */
export const FRAMEWORK_STACK_PACK: Readonly<Record<string, string>> = {
  "quarkus": "quarkus-patterns",
  "spring-boot": "spring-patterns",
  "nestjs": "nestjs-patterns",
  "express": "express-patterns",
  "fastapi": "fastapi-patterns",
  "django": "django-patterns",
  "gin": "gin-patterns",
  "ktor": "ktor-patterns",
  "axum": "axum-patterns",
  "dotnet": "dotnet-patterns",
  "click": "click-cli-patterns",
};

/** Return the knowledge pack directory name for a framework, or empty string. */
export function getStackPackName(framework: string): string {
  return FRAMEWORK_STACK_PACK[framework] ?? "";
}

File: java/src/main/java/dev/iadev/domain/stack/StackPackMapping.java
package dev.iadev.domain.stack;

import java.util.Map;

/**
 * Framework to knowledge pack mapping.
 *
 * <p>Maps framework names to their corresponding knowledge pack
 * directory names. Used during generation to select framework-specific
 * documentation and patterns.</p>
 *
 * <p>Zero external framework dependencies (RULE-007).</p>
 */
public final class StackPackMapping {

    private StackPackMapping() {
        // utility class
    }

    /** Framework name to knowledge pack directory name (11 entries). */
    public static final Map<String, String> FRAMEWORK_STACK_PACK = Map.ofEntries(
            Map.entry("quarkus", "quarkus-patterns"),
            Map.entry("spring-boot", "spring-patterns"),
            Map.entry("nestjs", "nestjs-patterns"),
            Map.entry("express", "express-patterns"),
            Map.entry("fastapi", "fastapi-patterns"),
            Map.entry("django", "django-patterns"),
            Map.entry("gin", "gin-patterns"),
            Map.entry("ktor", "ktor-patterns"),
            Map.entry("axum", "axum-patterns"),
            Map.entry("dotnet", "dotnet-patterns"),
            Map.entry("click", "click-cli-patterns")
    );

    /**
     * Returns the knowledge pack directory name for a framework.
     *
     * @param framework the framework name
     * @return the pack name, or empty string if not found
     */
    public static String getStackPackName(String framework) {
        return FRAMEWORK_STACK_PACK.getOrDefault(framework, "");
    }
}
